### PR TITLE
Gate AppHost dotnet run hook on Aspire CLI version

### LINKED ISSUE #18359: AppHost SDK dotnet run hook can recurse with older Aspire CLI on PATH
## Summary

Running a latest/main AppHost with the AppHost SDK `dotnet run` hook enabled can recursively spawn `aspire run` processes when the `aspire` command on PATH is an older stable CLI, such as 13.4.x.

The problem reproduces with version skew: newer AppHost SDK targets delegate `dotnet run` to `aspire run`, but the older CLI does not have the corresponding behavior needed to suppress the hook when it launches the AppHost again.

## Repro shape

1. Install/use Aspire CLI 13.4.x on PATH.
2. Use an AppHost built against latest/main or daily 13.5 AppHost SDK targets where `dotnet run` delegates to the Aspire CLI bundle.
3. From a file-based AppHost directory, run:

```powershell
dotnet apphost.cs
```

Observed result: repeated `aspire.exe` / `dotnet.exe` process spawning. The same scenario stops reproducing after updating the global Aspire CLI to daily 13.5.0-preview.1.26319.9.

## Root cause

The newer AppHost SDK target rewrites `dotnet run` for AppHosts to:

```text
aspire run --project <apphost> --no-build --
```

The SDK target already supports suppression via `ASPIRE_SUPPRESS_CLI_RUN_HOOK` / `_AspireSuppressCliRunHook`, but an older 13.4.x CLI does not set/pass that suppression when it internally launches a file-based AppHost via `dotnet run --file apphost.cs`.

That produces this loop:

```text
dotnet apphost.cs
  -> SDK hook delegates to aspire run
    -> old aspire CLI launches dotnet run --file apphost.cs
      -> SDK hook delegates to aspire run again
        -> ...
```

## Expected behavior

The SDK `dotnet run` delegation should only happen when the resolved Aspire CLI is new enough to provide the required reciprocal behavior. If the resolved CLI is older than the minimum required version, the SDK should not delegate `dotnet run` to `aspire run` and should fall back to the normal `dotnet run` behavior instead.

The proposed minimum version is Aspire CLI 13.5.

## Notes

This is not a pure latest/main-to-latest/main bug. It requires skew between a newer AppHost SDK/run hook and an older CLI on PATH. That is why it stopped reproducing after updating the global CLI to a daily 13.5 build.

### PR DESCRIPTION
**Pull request by DamianEdwards**
## Summary

- Add an AppHost SDK gate so `dotnet run` only delegates to `aspire run` when the resolved Aspire CLI is at least 13.5
- Fall back to normal `dotnet run` behavior when the CLI is older, unavailable, or has an unparseable version
- Add SDK target coverage for 13.4.x, 13.5 previews/stable, later versions, and unparseable output

Fixes #18359

## Testing

- `dotnet test --project tests/Aspire.Hosting.Sdk.Tests/Aspire.Hosting.Sdk.Tests.csproj --no-launch-profile -- --filter-not-trait "quarantined=true" --filter-not-trait "outerloop=true"`
- Installed PR build and manually verified a new starter app can be started from aspire cli & `dotnet run` when aspire CLI on PATH is 13.5 or 13.4

### CODE REVIEW

File: tests/Aspire.Hosting.Sdk.Tests/Aspire.Hosting.Sdk.Tests.csproj [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
   <PropertyGroup>
     <TargetFramework>$(DefaultTargetFramework)</TargetFramework>
+    <_AspireHostingTasksTfm Condition="'$(MSBuildRuntimeType)' == 'Core'">net8.0</_AspireHostingTasksTfm>
+    <_AspireHostingTasksTfm Condition="'$(MSBuildRuntimeType)' != 'Core'">net472</_AspireHostingTasksTfm>
```
[the remark this answers is not in the document]
Comment (DamianEdwards) [reply tests/Aspire.Hosting.Sdk.Tests/Aspire.Hosting.Sdk.Tests.csproj]: @copilot hmmm hard-coding this will mean it will need to be manually updated when the target TFM of the tasks project changes though. You still think this is the right trade-off?

[the remark this answers is not in the document]
Comment (DamianEdwards) [reply tests/Aspire.Hosting.Sdk.Tests/Aspire.Hosting.Sdk.Tests.csproj]: Switched to runtime discovery of the Tasks output: the csproj now bakes the Tasks output root (`AspireHostingTasksOutputDirectory`) and the helper picks the first non-`net4*` TFM subdirectory containing `Aspire.Hosting.Tasks.dll`. That fixes the desktop-MSBuild build case (the spawned `dotnet msbuild` always needs the Core build) without coupling the test project to a specific TFM string — if the Tasks project moves from `net8.0` to a newer TFM, the test picks it up automatically.

File: src/Aspire.Hosting.Tasks/GetAspireCliVersion.cs [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Microsoft.Build.Framework;
+
+namespace Aspire.Hosting.Tasks;
+
+/// <summary>
+/// Gets the version of the Aspire CLI executable.
+/// </summary>
+public sealed class GetAspireCliVersion : Microsoft.Build.Utilities.Task
```
Comment (davidfowl) [inline src/Aspire.Hosting.Tasks/GetAspireCliVersion.cs]: Can we use the ExecTask.

Comment (DamianEdwards) [reply src/Aspire.Hosting.Tasks/GetAspireCliVersion.cs]: Removed, moved the logic to MSBuild goop instead :smile:

**Comment by mitchdenny** [comment]
Thinking about the in-repo dev loop with this gate in place. Since `AspireCliPath` is empty for our `playground/*` and sample AppHosts, the gate probes `aspire --version` off the contributor's **global PATH** — which is decoupled from the branch/bits they're actually working on. Two outcomes, both a bit awkward:

- Global CLI ≥ 13.5 → `dotnet run` silently delegates to the **global** `aspire run`, so you're not exercising the freshly-built repo bits (orchestration metadata aside). Surprising when you're testing local hosting/CLI changes.
- Global CLI old/absent → falls back to plain `dotnet run`. Fine, but now behavior differs contributor-to-contributor based on whatever they happened to install.

Delegating to whatever's globally installed probably isn't what we want in-repo — but we also don't build the CLI as part of a normal repo build, so we can't unconditionally point `AspireCliPath` at a local build either.

Not blocking this PR (the gate is the right safety fix), but I think we should decide the intended in-repo story explicitly rather than letting it fall out of each person's global install. The direction I'd lean:

1. **Build the CLI as part of the playground loop and run it from artifacts.** Add a playground-level dependency (e.g. a build dependency / `ProjectReference` on `Aspire.Cli`) so running an in-repo AppHost triggers the CLI to build, then point `AspireCliPath` at the `artifacts/` output. That makes `dotnet run` self-consistent — you're always exercising the CLI from your branch, not whatever's globally installed.

2. **Isolate in-repo invocations with a worktree-specific `.aspire` folder.** The CLI already honors `ASPIRE_HOME` to relocate the `.aspire` home (hives, settings, bin, package caches). We could default that to a worktree-local `.aspire/` (gitignored) for in-repo runs so local experiments don't read/write the contributor's global `~/.aspire` state — and so parallel worktrees don't stomp on each other.

Net goal: in-repo `dotnet run` should deterministically exercise *this* worktree's CLI in an isolated environment, instead of silently depending on the global CLI version and global `~/.aspire` state. Curious what you think the default should be.

**Review by mitchdenny — APPROVED** [review]
Approving — the version gate is the right safety fix for the recursion (#18359), parsing handles the SemVer/stderr/timeout edge cases well, and the test matrix is thorough and hermetic. Left a non-blocking note about how we want the in-repo `dotnet run` loop to behave going forward (building the CLI from artifacts + a worktree-isolated `.aspire`).
